contracts/gateway/src/contract/execute.rs

```diff
@@ -1,158 +1,144 @@
-use std::collections::HashMap;
-
+use crate::contract::Error;
+use crate::events::GatewayEvent;
+use crate::router::Router;
+use crate::state;
+use crate::verifier::Verifier;
+use axelar_wasm_std::utils::TryMapExt;
 use axelar_wasm_std::VerificationStatus;
-use cosmwasm_std::{to_binary, Addr, DepsMut, Response, WasmMsg};
+use connection_router::state::Message;
+use cosmwasm_std::{Event, Response, Storage, WasmMsg};
 use error_stack::{Result, ResultExt};
 use itertools::Itertools;
 
-use crate::contract::query;
-use crate::contract::query::Verifier;
-use crate::error::ContractError;
-use connection_router::state::Message;
+pub fn verify_messages(verifier: &Verifier, msgs: Vec<Message>) -> Result<Response, Error> {
```

Comment:
I feel like there has been some back-and-forth regarding to the execute mod implementation with this PR now. Won't be able to mock store access anymore with this PR, which is the strength of the previous implementation?

Reply:
resolved during code walkthrough, mocking is still possible